DarkflameUniverse/DarkflameServer
# chore: make dNet only depend on dCommon libraries

- Add structs to communicate between the various functions so the data can be easily passed around to the correct handler
- Remove code that was dependent on other libraries linking against dNet by using the variables passed into the function instead, inlining the code to the one place it is used (A new issue should be created tracking this code so we can find a proper place for it later) and in one case inlining without a TODO or a struct due to the amount of data needed to be passed between the interfaces, and the fact that Character is a dGame class with multiple serialization types.

Tested that
- I can still load into worlds
- See other player positons update with and without mounts or racecars
- save data is in tact when delivered to my client
- bad words are rejected from civilians and gms can still bypass in global and friend chat
- remove cheat detection that isnt a cheat detection and doesn't really do anything anyways
- help issues are still the same as the strings in the inis
